sympy/tensor/array/expressions/array_expressions.py

```diff
@@ -44,7 +44,7 @@ def __getitem__(self, item):
         return ArrayElement(self, item)
 
     def as_explicit(self):
-        if any(not isinstance(i, (int, Integer)) for i in self.shape):
+        if not all(isinstance(i, (int, Integer)) for i in self.shape):
```

Comment:
This looks like something to be fixed elsewhere. The shape should really be required to be sympified or required to be an `int`. Allowing both is flakey.

Reply:
Indeed! I will not change it here though.